CONTRIBUTING.md

```diff
@@ -1,96 +1,232 @@
+*proposal*, *draft*, and *accepted* (or *rejected*), each of which has suggested
+entrance criteria and next steps detailed below. RFCs typically advance one
+stage at a time, but may advance multiple stages at a time. Stage
+advancements occur on GitHub.
+
+In general, it's preferable to start with a pull request so that we can best
+evaluate the RFC in detail. However, starting with an issue is also permitted if
+the full details are not worked out.
+
+All RFCs start as either a *strawman* or *proposal*.
+
+## Stage 0: *Strawman*
+
+An RFC at the *strawman* stage captures a described problem or
+partially-considered solutions. A *strawman* does not need to meet any entrance
+criteria. A *strawman's* goal is to prove or disprove a problem and guide
+discussion towards either rejection or a preferred solution. A *strawman* may
+be an issue or a pull request (though an illustrative pull request is preferrable).
+
+*There is no entrance criteria for a Strawman*
+
+As implied by the name [strawman](https://en.wikipedia.org/wiki/Straw_man_proposal),
+the goal at this stage is to knock it down (*reject*) by considering other
+possible related solutions, showing that the motivating problem can be solved
+with no change to the specification, or that it is not aligned with the
+*guiding principles*.
+
+Once determined that the *strawman* is compelling, it should seek the entrance
+criteria for *proposal*.
+
+
+## Stage 1: *Proposal*
+
+An RFC at the *proposal* stage is a solution to a problem with enough fidelity
+to be discussed in detail. It must be backed by a willing *champion*. A
+*proposal*'s goal is to make a compelling case for acceptance by describing
+both the problem and the solution via examples and spec edits. A *proposal*
+should be a pull request.
+
+*Entrance criteria:*
+
+* Identified *champion*
+* Clear explanation of problem and solution
+* Illustrative examples
+* Incomplete spec edits
+* Identification of potential concerns, challenges, and drawbacks
+
+A *proposal* is subject to the same discussion as a *strawman*: ensuring that it
+is well aligned with the *guiding principles*, is a problem worth solving, and
+is the preferred solution to that problem. A *champion* is not expected to have
+confidence in every detail at this stage and should instead focus on identifying
+and resolving issues and edge-cases. To better understand the technical
+ramifications of the *proposal*, a *champion* is encouraged to implement it in a
+AsyncAPI library.
+
+Most *proposals* are expected to evolve or change and may be rejected. Therefore,
+it is unwise to rely on a *proposal* in a production environment. AsyncAPI
+libraries *may* implement *proposals*, though are encouraged to not enable the
+*proposed* feature without explicit opt-in.
+
+
+## Stage 2: *Draft*
+
+An RFC at the *draft* stage is a fully formed solution. There is
+consensus the problem identified should be solved, and this particular solution
```

Comment:
This does not sound quite right to me 🤔 You mean the problem has been identified and agreed upon or?

Reply:
I'm not sure what is that you don't understand there 😅

Reply:
never mind, read it a couple of times and it makes sense now 🤷